hack/update-test-infra.sh

```diff
@@ -37,12 +37,16 @@ for repo in *; do
   git remote update -p
   git pull
   git checkout -b ${branch} upstream/master
-  dep ensure -update github.com/knative/test-infra
-  ./hack/update-deps.sh
-  [[ -z "$(git diff)" ]] && continue
-  git commit -a -m "Update test-infra to the latest version"
-  git push -u origin ${branch}
-  echo -e "\nCheck the PR created above, and make changes if necessary"
+  if [[ -f "Gopkg.lock" ]]; then
+    dep ensure -update github.com/knative/test-infra
+    ./hack/update-deps.sh
+    [[ -z "$(git diff)" ]] && continue
+    git commit -a -m "Update test-infra to the latest version"
+    git push -u origin ${branch}
+    echo -e "\nCheck the PR created above, and make changes if necessary"
+  else
+    echo -e "\nGopkg.lock not found, skip updating"
```

Comment:
If you find `go.mod`, update using `go get` and `go mod vendor`. This is the case for observability and client repos.

Reply:
so do `go get` and `go mod vendor` instead of `dep ensure`, right?

Reply:
Yes.

```
GO111MODULE=on go get -u github.com/knative/test-infra/scripts
GO111MODULE=on go mod vendor
```

Reply:
This command failed in `observability`:
```
GO111MODULE=on go get -u github.com/knative/test-infra/scripts
go: dmitri.shuralyov.com/html/belt@v0.0.0-20180602232347-f7d459c86be0: unrecognized import path "dmitri.shuralyov.com/html/belt" (https fetch: Get https://dmitri.shuralyov.com/html/belt?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/service/change@v0.0.0-20181023043359-a85b471d5412: unrecognized import path "dmitri.shuralyov.com/service/change" (https fetch: Get https://dmitri.shuralyov.com/service/change?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/app/changes@v0.0.0-20180602232624-0a106ad413e3: unrecognized import path "dmitri.shuralyov.com/app/changes" (https fetch: Get https://dmitri.shuralyov.com/app/changes?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/state@v0.0.0-20180228185332-28bcc343414c: unrecognized import path "dmitri.shuralyov.com/state" (https fetch: Get https://dmitri.shuralyov.com/state?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: error loading module requirements
```

I will take a look at what' going on

Reply:
it works now, will update the PR

Reply:
Updated